### PR TITLE
DAOS-9355 (doc): Add DAOS-9725 to 2.0.1 release_notes.md

### PR DESCRIPTION
**Pull request by Michael-Hennecke**
Add a referenct to DAOS-9725 (DER_CSUM) to the "known issues"
section of the 2.0.1 release notes.

Doc-only: true
Signed-off-by: Michael Hennecke <michael.hennecke@intel.com>

### CODE REVIEW

**Review by mchaarawi — DISMISSED** [review]

File: docs/release/release_notes.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 We are pleased to announce the release of DAOS version 2.0.
```
Comment (kprantis) [inline docs/release/release_notes.md:3]: Just noticed this, but don't we want this to say "2.0.1" versus "2.0"?

File: docs/release/release_notes.md [hunk 2]
```diff
@@ -32,6 +32,16 @@ The DAOS 2.0.1 release contains the following updates on top of DAOS 2.0.0:
...
+## Known Issues and limitations
+
+During SOAK testing of DAOS 2.0.1,
+some test jobs have failed with DER\_CSUM(-2021) checksum errors.
+This sighting is currently being investigated to determine if this is
+caused by a client-side issue with the checksum verification code,
+triggered by a hardware media error in the testing environment, or is
+a bug in the server code that may potentially cause a data corruption.
+See [DAOS-9725](https://daosio.atlassian.net/browse/DAOS-9725).
+
```
Comment (kprantis) [inline docs/release/release_notes.md:44]: @johannlombardi So you wanted us to restore the known issues that were in 2.0 but still persist in 2.0.1, correct?

Comment (johannlombardi) [reply docs/release/release_notes.md:44]: Right, there might be some other bugs we want to mention there.

File: docs/release/release_notes.md (hunk 1 above)
Comment (Michael-Hennecke) [reply docs/release/release_notes.md:3]: i'd keep this generic at 2.0, the patchlevels are on the headings following this toplevel section

File: docs/release/release_notes.md (hunk 2 above)
Comment (Michael-Hennecke) [reply docs/release/release_notes.md:64]: done

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**